simple/app.go

```diff
@@ -173,6 +168,24 @@ type BasicReconcileOptions struct {
 	UsePlain bool
 }
 
+type AppCustomRouteMethod string
+
+const (
+	AppCustomRouteMethodGet    AppCustomRouteMethod = "GET"
+	AppCustomRouteMethodPut    AppCustomRouteMethod = "PUT"
+	AppCustomRouteMethodPost   AppCustomRouteMethod = "POST"
+	AppCustomRouteMethodDelete AppCustomRouteMethod = "DELETE"
```

Comment:
I'm really sorry for not being clear in my comment, but we probably need to expand this list to cover all methods we expect to expose to developers.

I'm not sure if we need the likes of `HEAD` or `CONNECT` but we at least should include `PATCH`.

Maybe just copying what `net/http` has is enough:

```suggestion
	AppCustomRouteMethodConnect AppCustomRouteMethod = http.MethodConnect
	AppCustomRouteMethodDelete  AppCustomRouteMethod = http.MethodDelete
	AppCustomRouteMethodGet     AppCustomRouteMethod = http.MethodGet
	AppCustomRouteMethodHead    AppCustomRouteMethod = http.MethodHead
	AppCustomRouteMethodOptions AppCustomRouteMethod = http.MethodOptions
	AppCustomRouteMethodPatch   AppCustomRouteMethod = http.MethodPatch
	AppCustomRouteMethodPost    AppCustomRouteMethod = http.MethodPost
	AppCustomRouteMethodPut     AppCustomRouteMethod = http.MethodPut
	AppCustomRouteMethodTrace   AppCustomRouteMethod = http.MethodTrace
```